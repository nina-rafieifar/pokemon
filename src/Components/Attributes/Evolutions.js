import { html, LitElement, css } from "lit-element";
class PokemonEvolution extends LitElement {
  static get properties() {
    return {
      name: { attribute: true },
      number: { attribute: true },
      image: { attribute: true }
    };
  }
  static get styles() {
    return css`
      .pokemon__evolution_general-attributes {
        box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
        transition: 0.3s;
        border-radius: 50%;
        background-color: #ffffff;
        border: 1px solid #3b3939;
        height: 25vh;
        width: 25vh;
        color: #3b3939;
      }
      .pokemon__evolution_name {
        margin: 0;
        margin: -25px;
        font-size: small;
      }
      .pokemon__evolution_number {
        padding: 0.4rem 0px;
        font-family: Tigerious;
        margin: 0.1rem;
      }
      .pokemon__evolution_image {
        height: 12vh;
        pointer-events: none;
        text-align: left;
      }
    `;
  }
  render() {
    return html`<div class="pokemon__evolution_general-attributes">
      <p class="pokemon__evolution_number">${this.number}</p>
      <img alt=${this.name} src=${this.image} class="pokemon__evolution_image" />
      <h4 class="pokemon__evolution_name">${this.name}</h4>
    </div>`;
  }
}

customElements.define("pokemon-evolution", PokemonEvolution);
