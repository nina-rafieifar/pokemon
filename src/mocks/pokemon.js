export const pokemon = {
  attacks: { fast: [{ damage: 12, name: "Tackle" }], special: [{ damage: 70, name: "Power Whip" }] },
  evolutionRequirements: { amount: 25, name: "Bulbasaur candies" },
  evolutions: [
    {
      id: "UG9rZW1vbjowMDI=",
      number: "002",
      name: "Ivysaur",
      image: "https://img.pokemondb.net/artwork/ivysaur.jpg"
    },
    {
      id: "UG9rZW1vbjowMDM=",
      number: "003",
      name: "Venusaur",
      image: "https://img.pokemondb.net/artwork/venusaur.jpg"
    }
  ],
  height: { __typename: "PokemonDimension", minimum: "0.61m", maximum: "0.79m" },
  id: "UG9rZW1vbjowMDE=",
  image: "https://img.pokemondb.net/artwork/bulbasaur.jpg",
  name: "Bulbasaur",
  number: "001",
  resistant: ["Water", "Electric", "Grass", "Fighting", "Fairy"],
  weaknesses: ["Fire", "Ice", "Flying", "Psychic"],
  weight: { minimum: "6.04kg", maximum: "7.76kg" }
};
