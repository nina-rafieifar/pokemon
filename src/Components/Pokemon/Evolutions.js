import React from "react";
import KeyboardArrowRightIcon from "@material-ui/icons/KeyboardArrowRight";
import "../Attributes/Evolutions";
import "../Attributes/Sizes";
import "./style.scss";

export function Evolutions(prop) {
  return (
    <table className="pokemon__evolutions_table">
      <tbody>
        <tr>
          <td>
            <div className="pokemon__evolutions_container">
              <p>Evolution</p>
              {prop.pokemon.evolutions ? (
                <div className="pokemon__evolutions_items">
                  <pokemon-evolution
                    name={prop.pokemon.name}
                    number={prop.pokemon.number}
                    image={prop.pokemon.image}
                  ></pokemon-evolution>
                  <KeyboardArrowRightIcon />
                  <pokemon-evolution
                    name={prop.pokemon.evolutions[0].name}
                    number={prop.pokemon.evolutions[0].number}
                    image={prop.pokemon.evolutions[0].image}
                  ></pokemon-evolution>
                </div>
              ) : (
                <div className="pokemon__evolutions_items pokemon__evolutions_items_no-evolve">
                  <p>This Pok√©mon does not evolve.</p>
                  <div>
                    <pokemon-evolution
                      name={prop.pokemon.name}
                      number={prop.pokemon.number}
                      image={prop.pokemon.image}
                    ></pokemon-evolution>
                  </div>
                </div>
              )}
            </div>
          </td>
        </tr>
        <tr>
          <td className="pokemon__details_items">
            <pokemon-sizes
              weightMax={prop.pokemon.weight.maximum}
              weightMin={prop.pokemon.weight.minimum}
              heightMax={prop.pokemon.height.maximum}
              heightMin={prop.pokemon.height.minimum}
            ></pokemon-sizes>
          </td>
        </tr>
      </tbody>
    </table>
  );
}
