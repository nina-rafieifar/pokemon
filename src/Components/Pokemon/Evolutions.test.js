import React from "react";
import { render } from "@testing-library/react";
import { Evolutions } from "./Evolutions";
import { pokemon } from "../../mocks/pokemon";

test("renders Pokemon", () => {
  const element = render(<Evolutions pokemon={pokemon} />);
  expect(element).toMatchSnapshot();
});

test("renders Pokemon no evolve", () => {
  const noEvolvePokemon = pokemon;
  noEvolvePokemon.evolutions = null;
  const element = render(<Evolutions pokemon={noEvolvePokemon} />);
  expect(element).toMatchSnapshot();
});
