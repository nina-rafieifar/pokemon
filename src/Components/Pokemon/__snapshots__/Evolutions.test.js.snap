// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pokemon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="pokemon__evolutions_table"
      >
        <tbody>
          <tr>
            <td>
              <div
                class="pokemon__evolutions_container"
              >
                <p>
                  Evolution
                </p>
                <div
                  class="pokemon__evolutions_items"
                >
                  <pokemon-evolution
                    image="https://img.pokemondb.net/artwork/bulbasaur.jpg"
                    name="Bulbasaur"
                    number="001"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                    />
                  </svg>
                  <pokemon-evolution
                    image="https://img.pokemondb.net/artwork/ivysaur.jpg"
                    name="Ivysaur"
                    number="002"
                  />
                </div>
              </div>
            </td>
          </tr>
          <tr>
            <td
              class="pokemon__details_items"
            >
              <pokemon-sizes
                heightmax="0.79m"
                heightmin="0.61m"
                weightmax="7.76kg"
                weightmin="6.04kg"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="pokemon__evolutions_table"
    >
      <tbody>
        <tr>
          <td>
            <div
              class="pokemon__evolutions_container"
            >
              <p>
                Evolution
              </p>
              <div
                class="pokemon__evolutions_items"
              >
                <pokemon-evolution
                  image="https://img.pokemondb.net/artwork/bulbasaur.jpg"
                  name="Bulbasaur"
                  number="001"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                  />
                </svg>
                <pokemon-evolution
                  image="https://img.pokemondb.net/artwork/ivysaur.jpg"
                  name="Ivysaur"
                  number="002"
                />
              </div>
            </div>
          </td>
        </tr>
        <tr>
          <td
            class="pokemon__details_items"
          >
            <pokemon-sizes
              heightmax="0.79m"
              heightmin="0.61m"
              weightmax="7.76kg"
              weightmin="6.04kg"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders Pokemon no evolve 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="pokemon__evolutions_table"
      >
        <tbody>
          <tr>
            <td>
              <div
                class="pokemon__evolutions_container"
              >
                <p>
                  Evolution
                </p>
                <div
                  class="pokemon__evolutions_items pokemon__evolutions_items_no-evolve"
                >
                  <p>
                    This Pokémon does not evolve.
                  </p>
                  <div>
                    <pokemon-evolution
                      image="https://img.pokemondb.net/artwork/bulbasaur.jpg"
                      name="Bulbasaur"
                      number="001"
                    />
                  </div>
                </div>
              </div>
            </td>
          </tr>
          <tr>
            <td
              class="pokemon__details_items"
            >
              <pokemon-sizes
                heightmax="0.79m"
                heightmin="0.61m"
                weightmax="7.76kg"
                weightmin="6.04kg"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="pokemon__evolutions_table"
    >
      <tbody>
        <tr>
          <td>
            <div
              class="pokemon__evolutions_container"
            >
              <p>
                Evolution
              </p>
              <div
                class="pokemon__evolutions_items pokemon__evolutions_items_no-evolve"
              >
                <p>
                  This Pokémon does not evolve.
                </p>
                <div>
                  <pokemon-evolution
                    image="https://img.pokemondb.net/artwork/bulbasaur.jpg"
                    name="Bulbasaur"
                    number="001"
                  />
                </div>
              </div>
            </div>
          </td>
        </tr>
        <tr>
          <td
            class="pokemon__details_items"
          >
            <pokemon-sizes
              heightmax="0.79m"
              heightmin="0.61m"
              weightmax="7.76kg"
              weightmin="6.04kg"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
